### PR TITLE
feature: windows support, no deps, tests, CI(ubuntu, macos, windows)

### LINKED ISSUE #2: Code using pcre does not compile on macOS
Compilation error:

```
==================
              ^
/var/folders/2y/7ldjq9qn1ls6b184zfv7040m0000gn/T/v/main.13159573688693451185.tmp.c:1226:31: error: expected identifier or '('
static string struct pcre_str(struct pcre it); // auto
                              ^
/var/folders/2y/7ldjq9qn1ls6b184zfv7040m0000gn/T/v/main.13159573688693451185.tmp.c:1226:31: error: expected ')'
/var/folders/2y/7ldjq9qn1ls6b184zfv7040m0000gn/T/v/main.13159573688693451185.tmp.c:1226:30: note: to match this '('
static string struct pcre_str(struct pcre it); // auto
                             ^
/var/folders/2y/7ldjq9qn1ls6b184zfv7040m0000gn/T/v/main.13159573688693451185.tmp.c:1227:28: error: expected ';' after top level declarator
static string indent_struct pcre_str(struct pcre it, int indent_count); // auto
                           ^
                           ;
...
==================
(Use `v -cg` to print the entire error message)
```

I have pcre installed via Homebrew:

```
$ brew info pcre
pcre: stable 8.44 (bottled), HEAD
Perl compatible regular expressions library
https://www.pcre.org/
/usr/local/Cellar/pcre/8.44 (204 files, 5.5MB) *
  Poured from bottle on 2020-05-19 at 12:31:13
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/pcre.rb
License: BSD-3-Clause
==> Options
--HEAD
        Install HEAD version
==> Analytics
install: 168,466 (30 days), 553,389 (90 days), 2,721,188 (365 days)
install-on-request: 6,893 (30 days), 23,707 (90 days), 82,768 (365 days)
build-error: 0 (30 days)
```

I'm not familiar with C. Is this perhaps a C-compiler issue?

### PR DESCRIPTION
**Pull request by ghost**
Fixes #2

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.vscode
+.idea
+
```
Comment (JalonSolov) [inline .gitignore:4]: Needs a `*.iml` as well, for IDEA.

File: README.md [hunk 2]
```diff
@@ -1,19 +1,6 @@
 ## pcre
 
 A simple regex library for [V](https://github.com/vlang/v).
-It wraps the venerable [PCRE](https://www.pcre.org/) library, so
```
Comment (spytheman) [inline README.md:4]: The link to the original https://www.pcre.org/ should stay.

**Review by spytheman — CHANGES_REQUESTED** [review]
Good work, just put back the link to the original PCRE site into the README.

File: README.md (hunk 2 above)
Comment (ghost) [reply README.md:4]: There is a link in "Built With".

Comment (ghost) [reply README.md:4]: Keep it both?

Comment (spytheman) [reply README.md:4]: I did not see it.

Comment (spytheman) [reply README.md:4]: No, I think one is enough.

**Comment by tslmy** [comment]
Thank you for fixing #2 :D

**Pull request merged**